src/db.rs

```diff
@@ -121,150 +166,229 @@ pub fn get_connection(uuid: Uuid) -> Result<Connection> {
 }
 
 pub fn get_secret_policy(sec: &str) -> Option<policy::Policy> {
-    let mut dbconn = get_dbconn().unwrap();
+    let mut dbconn = get_dbconn().ok()?;
     let mut trnsx = dbconn.start_transaction(TxOpts::default()).ok()?;
 
-    let mqstr = "SELECT polids FROM secpol WHERE secret = ?";
-
-    let val: Option<i32> = trnsx.exec_first(mqstr, (sec,)).ok()?;
-
-    if val.is_none() {
-        info!("No policy has been set for secret with id: {}", &sec);
-        return None;
-    }
+    let mqstr = "SELECT polid FROM secrets WHERE secret_id = ?";
 
+    let val_row: mysql::Row = trnsx.exec_first(mqstr, (sec,)).ok()?.unwrap();
+    let val = match mysql::from_value_opt::<u64>(val_row["polid"].clone()) {
+        Ok(u) => u,
+        Err(_e) => return None,
+    };
+    /*
     let mqstr = "SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids FROM policy WHERE id = ?";
 
     let polval: Vec<(String, String, u32, u32, String)> = trnsx.exec(mqstr, (val,)).ok()?;
-
     Some(policy::Policy {
         allowed_digests: serde_json::from_str(&polval[0].0).unwrap(),
         allowed_policies: serde_json::from_str(&polval[0].1).unwrap(),
         min_fw_api_major: polval[0].2,
         min_fw_api_minor: polval[0].3,
         allowed_build_ids: serde_json::from_str(&polval[0].4).unwrap(),
     })
+    */
+    let secret_policy = get_policy(val).ok()?;
+    Some(secret_policy)
 }
 
-pub fn insert_polid(sec: &str, polid: u32) -> Result<u32> {
-    let mut dbconn = get_dbconn().unwrap();
+pub fn insert_keyset(ksetid: &str, kskeys: &[String], polid: u32) -> Result<()> {
+    let mut dbconn = get_dbconn()?;
 
-    let mqstr = "INSERT INTO secpol (secret, polids ) VALUES(?, ?)".to_string();
+    let mysqlstr = "INSERT INTO keysets (keysetid, kskeys, polid)
+                    VALUES(?, ?, ?)"
+        .to_string();
 
     let mut trnsx = dbconn.start_transaction(TxOpts::default())?;
+    /* create JSON for vector struct member variables */
+    let kskeys_str = serde_json::to_string(kskeys)?;
 
-    trnsx.exec_drop(mqstr, (&sec, &polid))?;
-    trnsx.commit()?;
-    Ok(polid)
+    trnsx.exec_drop(mysqlstr, (ksetid, &kskeys_str, polid))?;
+
+    Ok(trnsx.commit()?)
```

Comment:
So I think you can call `dbconn.exec_drop()` directly, no need for transaction (I think transaction should only be used when you have multiple statements you want to group together and commit atomically).

Reply:
I verified commented out code is gone from my local copy and will push a new version.  I hear what you are saying and agree that for small numbers of transactions, exec_drop and query_drop will work without a commit.  I would like to keep the commits because we use INNODB on our gensev2 copy of mysql, then start a transaction, with query_drop and exec_drop.  If we don't commit, it is possible that we could miss committing a change when multiple connections are trying to execute the same rust function.  I am sure that this will work with INNODB autocommits, but then we have to assume that everyone will set up their db's with autocommits turned on.   Is there a reason why keeping the commit presents a problem?